src/kernels/jupyter/launcher/serverUriStorage.ts

```diff
@@ -28,16 +31,32 @@ export class JupyterServerUriStorage implements IJupyterServerUriStorage {
     public get onDidRemoveUris() {
         return this._onDidRemoveUris.event;
     }
+    public get currentUri(): string | undefined {
+        return this._currentUri;
+    }
+    public get onDidChange(): Event<void> {
+        return this._onDidChangeUri.event;
+    }
+    public get isLocalLaunch(): boolean {
+        return this.currentUri === Settings.JupyterServerLocalLaunch || this.currentUri === undefined;
+    }
     constructor(
         @inject(IWorkspaceService) private readonly workspaceService: IWorkspaceService,
         @inject(ICryptoUtils) private readonly crypto: ICryptoUtils,
         @inject(IEncryptedStorage) private readonly encryptedStorage: IEncryptedStorage,
         @inject(IApplicationEnvironment) private readonly appEnv: IApplicationEnvironment,
         @inject(IMemento) @named(GLOBAL_MEMENTO) private readonly globalMemento: Memento,
-        @inject(ServerConnectionType) private readonly serverConnectionType: ServerConnectionType
+        @inject(IsWebExtension) readonly isWebExtension: boolean
     ) {
+        // Remember if local only
+        this._localOnly = isWebExtension
+            ? false
+            : this.globalMemento.get<boolean>(mementoKeyToIndicateIfConnectingToLocalKernelsOnly, true);
+
         // Cache our current state so we don't keep asking for it from the encrypted storage
-        this.getUri().ignoreErrors();
+        this.getUri()
+            .then((s) => (this._currentUri = s))
```

Comment:
I don't think we can assume this will happen very quickly, its possible the encrypted storage is very slow (or could present UI dialog boxes to the user and the like) & the rest of the code will run to completion, leaving the properties like `currentUri` & `isLocalLaunch` in a bad (incorect state).

E.g. in the past when ever we accessed encrypted storage, we'd get a dialog prompting the user to allow accessing secret store, this could happen in the future as well, (.e. this promise could get resolved seconds or minutes later)

Reply:
along with a private property (so that we have sync & latest access to information - even while we're still busy updating the memeto)

Reply:
Actually isLocalLaunch should use the _local flag. That was a copy of the logic in the ServerConnectionType. I'll update.